src/services/ContributorAuditService/providers/Bitbucket/BitbucketContributorAuditProvider.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  IContributorAuditModel,
+  IContributorAuditRepositories,
+} from "../../../../api/SOOSHooksApiClient";
+import { soosLogger } from "../../../../logging";
+import { IContributorAuditProvider } from "../../ContributorAuditService";
+import { DataMappingUtilities, ParamUtilities } from "../../utilities";
+import BitbucketApiClient, { BitbucketRepository } from "./BitbucketApiClient";
+import { SOOS_BITBUCKET_CONTRIBUTOR_AUDIT_CONSTANTS } from "./constants";
+
+class BitbucketContributorAuditProvider implements IContributorAuditProvider {
+  public async audit(
+    implementationParams: Record<string, string | number>,
+  ): Promise<IContributorAuditModel> {
+    const bitbucketPAT = ParamUtilities.getAsString(implementationParams, "secret");
+    const organizationName = ParamUtilities.getAsString(implementationParams, "organizationName");
+    const days = ParamUtilities.getAsNumber(implementationParams, "days");
+    const userName = ParamUtilities.getAsString(implementationParams, "userName");
+    const bitbucketApiClient = new BitbucketApiClient(
+      days,
+      userName,
+      bitbucketPAT,
+      organizationName,
+    );
+    const repositories = await bitbucketApiClient.getBitbucketRepositories();
+    soosLogger.verboseDebug("Fetching commits for each repository");
+    const contributors = await this.getBitbucketRepositoryContributors(
+      bitbucketApiClient,
+      repositories,
+      SOOS_BITBUCKET_CONTRIBUTOR_AUDIT_CONSTANTS.RequestBatchSize,
+    );
+    const scriptVersion = ParamUtilities.getAsString(implementationParams, "scriptVersion");
+
+    const finalContributors: IContributorAuditModel = {
+      metadata: {
+        scriptVersion: scriptVersion,
+        days: days,
+      },
+      organizationName: organizationName,
+      contributors: contributors,
+    };
+
+    return finalContributors;
+  }
+
+  public validateParams(implementationParams: Record<string, string | number>): void {
+    if (!implementationParams["secret"]) {
+      throw new Error(
+        `A Bitbucket App passsword is required as the '--secret', learn more at ${SOOS_BITBUCKET_CONTRIBUTOR_AUDIT_CONSTANTS.Urls.Docs.AppPassword}`,
+      );
+    }
+    if (!implementationParams["userName"]) {
+      throw new Error(`A Bitbucket username is required as the '--userName'`);
+    }
+  }
+
+  private async getBitbucketRepositoryContributors(
```

Comment:
a lot looks copied from the github implementation - can we put functions like this into a shared utilty?

Reply:
Consolidated what I could on `DataMappingUtilities`  for common interfaces they use, not sure if it could be simplified more without over complicating things. Probably could be worth it if we add support for more scm and we start repeating even more code